libibverbs/verbs.c

```diff
@@ -695,23 +695,26 @@ LATEST_SYMVER_FUNC(ibv_query_qp, 1_1, "IBVERBS_1.1",
 int ibv_query_qp_data_in_order(struct ibv_qp *qp, enum ibv_wr_opcode op,
 			       uint32_t flags)
 {
+	int result;
+
+	if (!check_comp_mask(flags, IBV_QUERY_QP_DATA_IN_ORDER_RETURN_CAPS))
```

Comment:
This function sure has become weird, flags is less of a flags and more of a selector now..

Still, flags has to be validated.. It can only be 0, IBV_QUERY_QP_DATA_IN_ORDER_RETURN_CAPS or  IBV_QUERY_QP_DATA_IN_ORDER_DEVICE_ONLY, everything else is a failure and returns 0.